hydra-node/src/Hydra/HeadLogic/Outcome.hs

```diff
@@ -49,19 +49,22 @@ data StateChanged tx
       , chainState :: ChainStateType tx
       , headId :: HeadId
       , headSeed :: HeadSeed
+      , stateChangeID :: Word64
```

Comment:
Should: keep this identifier outside of the sum type

I would strongly suggest to create a wrapping type to not need to put this in each sum constructor and also match on all constructors to implement `getStateChangeID`. For example:

```haskell
data StateEvent tx = StateEvent { eventId :: EventId, stateChanged :: StateChanged tx }
```